frontend/src/Components/Layout/Navigation.js

```diff
@@ -14,12 +21,14 @@ function Navigation() {
       <Navbar.Collapse id="basic-navbar-nav">
         <Nav className="me-auto">
           <Nav.Link as={Link} to="/">Home</Nav.Link>
-          <Nav.Link as={Link} to="/myorders">My Orders</Nav.Link>
           <Nav.Link as={Link} to="/cartpage">Cart ({items.length})</Nav.Link>
-          <Nav.Link as={Link} to="/editprofile">Manage My Profile</Nav.Link>
-          <Nav.Link as={Link} to="/signup">Sign Up</Nav.Link>
-          <Nav.Link as={Link} to="/login">Log In</Nav.Link>
-          <Nav.Link as={Link} to="/">Log Out</Nav.Link>
+        </Nav>
+        <Nav>
+          {userState && <Nav.Link as={Link} to="/myorders">My Orders</Nav.Link> }
+          {userState &&  <Nav.Link as={Link} to="/editprofile">Manage My Profile</Nav.Link> }
+          {!userState &&  <Nav.Link as={Link} to="/signup">Sign Up</Nav.Link> }
+          {!userState && <Nav.Link as={Link} to="/login">Log In</Nav.Link> }
+          {userState && <Nav.Link onClick={userDispatch}>Log Out</Nav.Link> }
```

Comment:
Refreshing the page logs you out. In other words, it resets the state. To implement this, simply replace onClick with href=<the page you want to get to>.